framework/db/Connection.php

```diff
@@ -337,6 +337,13 @@ class Connection extends Component
      * @see masterConfig
      */
     public $masters = [];
+
+    /**
+     * @var string the class name of the [[command]] object.
```

Comment:
@SilverFire I don't think we should make it too dynamic here; it will require me to duplicate more logic. Instead if you want customization you can set it via the DI container and it'll work. 
Does that sound like a good solution?